evap/results/exporters.py

```diff
@@ -1,68 +1,76 @@
-        self.sheet = None
-        self.col = 0
-        self.row = 0
+        super().__init__()
+
+        for index, color in self.COLOR_MAPPINGS.items():
+            self.workbook.set_colour_RGB(index, *color)
 
-    def normalize_number(self, number):
-        """ floors 'number' to a multiply of self.STEP """
+    @classmethod
+    def grade_to_style(cls, grade):
+        return 'grade_' + str(cls.normalize_number(grade))
+
+    @classmethod
+    def normalize_number(cls, number):
+        """ floors 'number' to a multiply of cls.STEP """
         rounded_number = round(number, 1)  # see #302
-        return round(int(rounded_number / self.STEP + 0.0001) * self.STEP, 1)
+        return round(int(rounded_number / cls.STEP + 0.0001) * cls.STEP, 1)
 
-    @staticmethod
-    def create_color(workbook, color_name, palette_index, color):
-        xlwt.add_palette_colour(color_name, palette_index)
-        workbook.set_colour_RGB(palette_index, *color)
-
-    def create_style(self, base_style, style_name, color_name):
-        self.styles[style_name] = xlwt.easyxf(base_style.format(color_name), num_format_str="0.0")
-
-    def init_styles(self, workbook):
-        self.styles = {
-            'default':          xlwt.Style.default_style,
-            'headline':         xlwt.easyxf('font: bold on, height 400; alignment: horiz centre, vert centre, wrap on; borders: bottom medium', num_format_str="0.0"),
-            'evaluation':       xlwt.easyxf('alignment: horiz centre, wrap on, rota 90; borders: left medium, top medium, right medium, bottom medium'),
-            'total_voters':     xlwt.easyxf('alignment: horiz centre; borders: left medium, right medium'),
-            'evaluation_rate':  xlwt.easyxf('alignment: horiz centre; borders: left medium, bottom medium, right medium'),
-            'evaluation_weight': xlwt.easyxf('alignment: horiz centre; borders: left medium, right medium'),
-            'bold':             xlwt.easyxf('font: bold on'),
-            'italic':           xlwt.easyxf('font: italic on'),
-            'border_left_right': xlwt.easyxf('borders: left medium, right medium'),
-            'border_top_bottom_right': xlwt.easyxf('borders: top medium, bottom medium, right medium'),
-            'border_top':       xlwt.easyxf('borders: top medium'),
-            'degree':           xlwt.easyxf('alignment: wrap on; borders: left medium, right medium'),
-        }
+    @classmethod
+    def init_grade_styles(cls):
+        """
+        Adds the grade styles to cls.styles and as a xlwt identifier.
+        This also notes all registered colors in cls.COLOR_MAPPINGS for the instances.
+
+        This method should only be called once, right after the class definition.
+        Instances need the styles, but they should only be registered once for xlwt.
+        """
+
+        if len(cls.COLOR_MAPPINGS) > 0:
+            # Method has already been called (probably in another import of this file).
+            return
```

Comment:
Didn't know about this, very cool. I don't know if there should be a warning, since it is okay, that the file is imported in multiple places, right? On the other hand, I am not sure, whether the file is "run" multiple times anyway (I tried with some `print` statements, but only got one line in the console output). I guess I can add an `ImportWarning`, since this is about importing and these warnings are ignored by default.
What do you think?

Reply:
With the current code, even with multiple imports, this code should only be run once.
Afaik, all warnings are ignored by default.
`ImportWarning` sounds good.